src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
+    {
+        $class = new \ReflectionClass(get_called_class());
+
+        return preg_replace('#Plugin$#', '', $class->getShortName());
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return array
+     */
+    final public static function getConfiguration($configuration = array())
+    {
+        return array_merge_recursive(
+            $configuration,
+            array(
+               'commands'   => static::getCommands(),
+               'presenters' => static::getPresenters(),
+               'matchers'   => static::getMatchers(),
```

Comment:
Seems fair, even though (still on a (quite) subjective object pov) I don't know if there is really any interest for the thing that registers the plugin to have any access to other methods